02_rand_dataset.ipynb

```diff
@@ -33,56 +24,400 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 68,
    "metadata": {
     "collapsed": false
    },
    "outputs": [],
    "source": [
     "# To get you started we can import Pandas and Seaborn which might help you\n",
     "# build a graph or visualisation of the data\n",
+    "% matplotlib inline\n",
+    "\n",
+    "from collections import defaultdict, Counter\n",
     "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib as mpl\n",
     "import pandas as pd\n",
     "import seaborn as sns\n",
     "import spacy\n",
     "\n",
     "nlp = spacy.load('en')\n",
     "\n",
-    "def read_file(file_name):\n",
+    "def read_file_to_list(file_name):\n",
     "    with open(file_name, 'r') as file:\n",
-    "        return file.read().decode('utf-8')\n",
+    "        return file.readlines()\n",
+    "\n",
+    "# The file has been re-encoded in UTF-8, the source encoding is Latin-1\n",
+    "terrorism_articles = read_file_to_list('data/rand-terrorism-dataset.txt')\n",
+    "\n",
+    "# Create a list of spaCy Doc objects representing articles\n",
+    "terrorism_articles_nlp = [nlp(art) for art in terrorism_articles]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example output using the full ~40,000 article dataset\n",
+    "\n",
+    "![Heatmap of terrorist group and country](images/example_output_full.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Example solution"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define some geographical areas and groups to inspect\n",
+    "\n",
+    "These are commonly mentioned in the full dataset, which you can prove for yourself using the same approaches as in the previous tutorial for Pride and Prejudice. You can process the full dataset at once into one spaCy span using the read_file() function. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 69,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "common_terr = ['taliban', \n",
```

Comment:
Formatting here is not correct. If you want to have every entry in the list in it's own line, correct formatting would be:
```

common_terr = [
    'taliban', 
    'al - qaeda', 
    'hamas',  
]

common_locs = [
    'iraq',
    'baghdad', 
    'kirkuk', 
]
```